### ISSUE TITLE
unable to use filter with levelname

### ISSUE THREAD

**Issue by michael-123123** [pytest-dev/pytest#11194]
Hello,

I have an issue where log filters are not working with level names. I looked for similar issues / solutions and couldn't find any.

A minimal example:

`pytest.ini` file:
```
[pytest]
log_cli = 1
log_cli_level = DEBUG
log_cli_format = %(levelname)-10s %(message)s
```

Minimal code snippet `test_example.py`
```
import logging


logger = logging.getLogger(__name__)


def log_filter(record: logging.LogRecord) -> bool:
    record.levelname = '[%s]' % record.levelname
    return True


logger.addFilter(log_filter)


def test_logger():
    logger.info('hello')
```

Expected output:
```
[INFO]       hello
```

Actual output:
```
INFO       hello
```

OS: Ubuntu 22.04
pytest version: 7.2.1
[pip_list.txt](https://github.com/pytest-dev/pytest/files/12010933/pip_list.txt)

**Comment by fifamessi** [comment]
why not 
`[pytest]
log_cli = 1
log_cli_level = DEBUG
log_cli_format = [%(levelname)s]          %(message)s`

**Comment by fifamessi** [comment]
I believe that in Python logging, formatters are applied after filters.
A naive solution would be (giving no access to pytest.ini or no desire to change it for the project):
```
import logging
import pytest

@pytest.fixture(scope='session', autouse=True)
def patch_root_logger_formater():
    for h in logging.getLogger('root').handlers:
        h.setFormatter(logging.Formatter("[%(levelname)s] %(message)s"))

@pytest.fixture(name='logger', scope='session')
def logger_fixture():
    return logging.getLogger(__name__)

def test_logger(logger):
    logger.info('hello')  
```

**Comment by michael-123123** [comment]
Hi @fifamessi 

thanks for the suggestion - but my point was that i don't want to configure the logger with code.
i want to add a filter (to some) of the handlers that only adds a '[]' wrapping around the levelname.

moreover, the way that the formatter works, afaik, is that it takes the values stored in the log record.
if a filter altered that value - for example added a '[]' wrapping around the level name - then it should be taken as is.

iiuc - there is a hardcoded lookup of level names somewhere in pytest, that overrides the content of the log message.

either way - i found alternative solutions outside of pytest for this.

**Comment by RonnyPfannschmidt** [comment]
Note that changing the level name at the record level is possibly breaking 3rd parties as some use it for lookup 

I consider expecting that to work consistently as a very ill advised expectation

**Comment by RonnyPfannschmidt** [comment]
https://github.com/pytest-dev/pytest/blob/main/src%2F_pytest%2Flogging.py#L127 handles the lookup in pytest 

I strongly recommend adding the brackets in the formatter configuration

**Comment by michael-123123** [comment]
@RonnyPfannschmidt thanks.
I ended up doing neither - and going with something else entirely.
My only point is that this seems like a bug.
Also, for my use case - it didn't really matter whether it breaks 3rd party libs.
thanks.

**Comment by RonnyPfannschmidt** [comment]
The hack in pytest is a very acceptable shortcut given the expectations one can have for logging records 

IMHO it's absolutely acceptable to use the apis as pytest- does 

If you change expected/implicit invariants, surprising behavior is to be excepted

**Comment by michael-123123** [comment]
I disagree. As you say - it's a hack. And the names of log levels are not afaik invariant.
Consider for example, a use case where someone wants log messages not in english.

But honestly - it doesn't matter to me all that much.
If this is not considered a bug - then this issue can be closed. 

I came across this while i tried to do something with pytest - and opened an issue.
In the meanwhile - I solved my problem another way.
